magento/magento2
# magento/magento2#12817: [Forwardport] Coupon code with canceled order.

Forwardport for https://github.com/magento-engcom/magento2ce/pull/1095


### Description

The number of coupon code uses is decreased if an order, that previously used this coupon code, is canceled.

### Fixed Issues (if relevant)

1. magento/magento2#12817: Coupon code with canceled order

### Manual testing scenarios

1. Create a Cart Price Rule with Specific Coupon. Set Uses per Coupon to 1.
2. Create Order (1st Order).
3. Apply the Coupon to 1st Order.
4. Place the 1st Order.
5. Create Order (2nd Order).
6. Apply the Coupon to 2nd Order.
7. Message "Coupon code is not valid" appears.
8. Cancel 1st Order.
9. Apply the Coupon to 2nd Order.
10. Message "Your coupon was successfully applied." appears.
11. Create a Cart Price Rule with Specific Coupon. Set Uses per Customer to 1.
12. Repeat steps 2-10 with this Coupon (note than orders must be placed by logged customer). 

### Contribution checklist
